File: frontend/src/types/timetable.ts
export interface TimetableOpening {
  name: string;
  openingTime: string;
  closingTime: string;
}

export interface Timetable {
  id: string;
  name: string;
  description: string | null;
  ownerId: string;
  openings: TimetableOpening[];
  createdAt: string;
  updatedAt: string;
}

export interface CreateTimetableRequest {
  name: string;
  description?: string;
  ownerId: string;
  openings: TimetableOpening[];
}

export interface UpdateTimetableRequest {
  name: string;
  description?: string;
  openings: TimetableOpening[];
}

File: src/Inventory.API/Data/Entities/Timetable.cs
namespace Inventory.API.Data.Entities;

public sealed class Timetable
{
    public string Id { get; set; } = "";
    public string Name { get; set; } = "";
    public string? Description { get; set; }
    public string OwnerId { get; set; } = "";
    public string Openings { get; set; } = "[]";
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }
}

public sealed class TimetableOpening
{
    public string Name { get; set; } = "";
    public string OpeningTime { get; set; } = "";
    public string ClosingTime { get; set; } = "";
}
